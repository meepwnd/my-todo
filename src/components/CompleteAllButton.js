import React from "react";
import { connect } from "react-redux";
import { completeAll } from "../actions/actions";

export const CompleteAllButton = ({ completeAll }) => (
  <button className="btn-complete-all" onClick={completeAll}>
    Complete All
  </button>
);

export default connect(
  null,
  { completeAll }
)(CompleteAllButton);
