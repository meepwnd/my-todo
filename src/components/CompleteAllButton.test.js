import React from 'react';
import {shallow} from 'enzyme'
import {CompleteAllButton} from './CompleteAllButton'

describe('clear all completed button', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<CompleteAllButton />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call clear all completed function on button click', () => {
    const completeAll = jest.fn()
    const wrapper = shallow(<CompleteAllButton completeAll={completeAll}/>)
    wrapper.find('button').simulate('click')
    expect(completeAll).toHaveBeenCalled()
  })
})