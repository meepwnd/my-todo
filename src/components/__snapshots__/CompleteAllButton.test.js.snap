// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clear all completed button should render correctly 1`] = `
<button
  className="btn-complete-all"
>
  Complete All
</button>
`;
